packages/agent/app/agent.ts

```diff
@@ -2,8 +2,17 @@ import { parse } from 'bowser';
 import { QueryParams } from './query-params';
 import { Agent } from '../shared/agent';
 import { run } from './runner';
+import { Controller, Operation, OperationFn, Sequence } from 'effection';
 
-export function* createAgent(queryParams: QueryParams) {
+// union of values that is yielded from createAgent
+type CreateAgentValue =
+  | Operation<Agent>
+  | OperationFn<void>
+  | Sequence<void> 
+  | PromiseLike<void> 
+  | Controller<void>;
+
+export function* createAgent(queryParams: QueryParams): Generator<CreateAgentValue, void, Agent> {
```

Comment:
So for example, here we would make this `Operation<Agent>`
```suggestion
export function* createAgent(queryParams: QueryParams): Operation<Agent> {
```